source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -12,65 +12,65 @@
+  readonly securityGroupIds?: string[]
+}
+
+export function buildElasticSearch(scope: Construct, props: BuildElasticSearchProps): [elasticsearch.CfnDomain, iam.Role] {
+
+  let subnetIds: string[] = [];
+  const constructDrivenProps: any = {};
 
   // Setup the IAM Role & policy for ES to configure Cognito User pool and Identity pool
-  const cognitoKibanaConfigureRole = new iam.Role(scope, 'CognitoKibanaConfigureRole', {
-    assumedBy: new iam.ServicePrincipal('es.amazonaws.com')
-  });
-
-  const cognitoKibanaConfigureRolePolicy = new iam.Policy(scope, 'CognitoKibanaConfigureRolePolicy', {
-    statements: [
-      new iam.PolicyStatement({
-        actions: [
-          "cognito-idp:DescribeUserPool",
-          "cognito-idp:CreateUserPoolClient",
-          "cognito-idp:DeleteUserPoolClient",
-          "cognito-idp:DescribeUserPoolClient",
-          "cognito-idp:AdminInitiateAuth",
-          "cognito-idp:AdminUserGlobalSignOut",
-          "cognito-idp:ListUserPoolClients",
-          "cognito-identity:DescribeIdentityPool",
-          "cognito-identity:UpdateIdentityPool",
-          "cognito-identity:SetIdentityPoolRoles",
-          "cognito-identity:GetIdentityPoolRoles",
-          "es:UpdateElasticsearchDomainConfig"
-        ],
-        resources: [
-          options.userpool.userPoolArn,
-          `arn:aws:cognito-identity:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:identitypool/${options.identitypool.ref}`,
-          `arn:aws:es:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:domain/${domainName}`
-        ]
-      }),
-      new iam.PolicyStatement({
-        actions: [
-          "iam:PassRole"
-        ],
-        conditions: {
-          StringLike: { 'iam:PassedToService': 'cognito-identity.amazonaws.com' }
-        },
-        resources: [
-          cognitoKibanaConfigureRole.roleArn
-        ]
-      })
-    ]
-  });
-  cognitoKibanaConfigureRolePolicy.attachToRole(cognitoKibanaConfigureRole);
+  const cognitoKibanaConfigureRole = createKibanaCognitoRole(scope, props.userpool, props.identitypool, props.domainName);
+
+  if (props.vpc) {
+    subnetIds = retrievePrivateSubnetIds(props.vpc);
+
+    if (subnetIds.length > MaximumAzsInElasticsearchDomain) {
+      subnetIds = subnetIds.slice(0, MaximumAzsInElasticsearchDomain);
+    }
 
-  let _cfnDomainProps = DefaultCfnDomainProps(domainName, cognitoKibanaConfigureRole, options);
+    constructDrivenProps.vpcOptions = {
+      subnetIds,
+      securityGroupIds: props.securityGroupIds
+    };
 
-  _cfnDomainProps = consolidateProps(_cfnDomainProps, cfnDomainProps);
+    // If the client did not submit a ClusterConfig, then we will create one
+    if (!props.clientDomainProps?.elasticsearchClusterConfig) {
+      constructDrivenProps.elasticsearchClusterConfig = createClusterConfiguration(subnetIds.length);
+    }
+  } else {
+    // If the client did not submit a ClusterConfig, then we will create one based on the Region
```

Comment:
Put something here to indicate this is the no VPC section, it's easy to miss that because the VPC section is so big and this is so short. perhaps:

```
} else { // No VPC
```